armi/reactor/components/component.py

```diff
@@ -594,7 +594,7 @@ def getNumberDensity(self, nucName):
 
     def getNuclideNumberDensities(self, nucNames):
         """Return a list of number densities for the nuc names requested."""
-        return [self.getNumberDensity(nucName) for nucName in nucNames]
+        return [self.p.numberDensities.get(nucName, 0.0) for nucName in nucNames]
```

Comment:
If calling `getNumberDensity` is discouraged, and `getNumberDensity` does the same as this line, should we just remove it?

Reply:
`getNumberDensity` here is a specific implementation of the `composite` pattern's broader one so it has to stay so that a component looks like a block/assembly/core/reactor/whatever via the api